docs/investigations/minimal-api-server.md

```diff
@@ -0,0 +1,55 @@
+# Minimal API Server
+
+The Kubernetes API machinery provides a pattern for declarative config-driven API with a number of conventions that simplify building configuration loops and consolidating sources of truth. There have been many efforts to make that tooling more reusable and less dependent on the rest of the Kube concepts but without a strong use case driving separation and a design the tooling is still fairly coupled to Kube.
+
+## Goal: 
+
+Running an API server that leverages the Kubernetes API server codebase, client libraries, and extension mechanisms without some or any of the built-in Kubernetes types either embedded or standalone. It is expected that most use cases would embed as a library vs be standalone.
+
+As a secondary goal, identifying where exceptions or undocumented assumptions exist in the libraries that would make clients behave differently generically (where an abstraction is not complete) should help ensure future clients can more concretely work across different API servers consistently.
+
+
+### Constraint: The abstraction for a minimal API server should not hinder Kubernetes development
+
+The primary consumer of the Kube API is Kubernetes - any abstraction that makes a standalone API server possible must not regress Kubernetes performance or overly complicate Kubernetes evolution. The abstraction *should* be an opportunity to improve interfaces within Kubernetes to decouple components and improve comprehension.
+
+### Constraint: Reusing the existing code base
+
+While it is certainly possible to rebuild all of Kube from scratch, a large amount of client tooling and benefit exists within patterns like declarative apply. This investigation is scoped to working within the context of improving the existing code and making the minimal changes within the bounds of API compatibility to broaden utility.
+
+It should be possible to include an arbitrary set of the existing Kube resource infra within the tooling so as to leverage things like `Namespaces` (for logical clusters) or `Secrets` (for benefitting from encryption at rest).
```

Comment:
I'm not sure I understand what this is trying to say. Can you elaborate?

Reply:
yeah, in brief if you want full fidelity of a resource behavior crds don’t cut it.  so secret encryption at rest or stringData behavior.  will expand a bit although would prefer a higher level use case.